tools/analyze-chain.py

```diff
@@ -49,59 +49,74 @@ def run_gen(env_data: bytes, block_program_args: bytes, flags: int):
         return 117, None, 0
 
 
+def generator_references_from_indices(c, height_to_hash_cache, indices: List[int]) -> List[bytes]:
+    ref_block_blobs = []
+    for h in indices:
+        if h not in height_to_hash_cache:
+            ref = c.execute("SELECT header_hash FROM full_blocks WHERE height=?", (h,))
+            height_to_hash_cache[h] = ref.fetchone()[0]
+            ref.close()
+
+        ref = c.execute("SELECT block FROM full_blocks WHERE header_hash=?", (height_to_hash_cache[h],))
```

Comment:
I believe you can just change the condition here to be `height=? AND in_main_chain=1` (and pass in `h`). Then you won't need the height-to-hash cache

Reply:
Seems to work, seems to be fast still, so this is a good change. Greatly simplifies the code.